src/prognostic_equations/edmfx_boundary_condition.jl

```diff
@@ -57,17 +165,56 @@ function approximate_inverf(x::FT) where {FT}
     return sign(x) * sqrt(term3 - term1)
 end
 
-function guass_quantile(p)
-    return sqrt(2) * approximate_inverf(2p - 1)
+"""
+    gauss_quantile(p::FT) where {FT}
+
+Computes the quantile (inverse of the cumulative distribution function, CDF)
+for a standard normal distribution N(0,1) at probability `p`.
+
+It uses the relationship Φ⁻¹(p) = √2 * erf⁻¹(2p - 1), where Φ⁻¹ is the
+normal quantile function and erf⁻¹ is the inverse error function, approximated
+by `approximate_inverf`.
+
+Arguments:
+- `p`: The probability (scalar or array element), ranging from 0 to 1.
+
+Returns:
+- The standard normal quantile corresponding to `p`.
+"""
+function gauss_quantile(p::FT) where {FT}
+    return sqrt(FT(2)) * approximate_inverf(2p - 1)
 end
 
+"""
+    percentile_bounds_mean_norm(low_percentile::FT, high_percentile::FT) where {FT}
+
+Calculates the mean value of a standard normal variable X ~ N(0,1) that is
+truncated to the interval `[xp_low, xp_high]`, where `xp_low` and `xp_high`
+are the quantiles corresponding to `low_percentile` and `high_percentile`
+respectively.
+
+The formula used is: E[X | xp_low ≤ X ≤ xp_high] = (ϕ(xp_low) - ϕ(xp_high)) / (P_high - P_low),
+where ϕ is the PDF of N(0,1), and P_high/P_low are the high/low percentiles.
+
+This gives a coefficient representing the expected value of a fluctuation
+drawn from a specific segment of a Gaussian distribution.
+
+Arguments:
+- `low_percentile`: The lower percentile bound (e.g., 0.8 for the 80th percentile).
+- `high_percentile`: The upper percentile bound (e.g., 0.9 for the 90th percentile).
+
+Returns:
+- The mean of the standard normal distribution truncated between the quantiles
+  of `low_percentile` and `high_percentile`.
+"""
 function percentile_bounds_mean_norm(
     low_percentile::FT,
-    high_percentile,
+    high_percentile::FT,
```

Comment:
I this @dennisYatunin was suggesting not to have extra FTs in functions?

Reply:
In general, it is better to avoid type restrictions like this for autodiff, since the types of certain arguments can change depending on whether they also encode partial derivatives. Most functions can extract floating point types from the parameter set (`FT = eltype(params)`), and there is no benefit to restricting the types of their other arguments. In this case, though, the parameter set is unavailable, so you need to make a choice of whether to grab `FT` from `low_percentile` or `high_percentile`. I can't imagine a situation where `low_percentile` will contain partial derivatives and `high_percentile` won't contain derivatives, or vice versa, so here it is okay to restrict their types to the same `FT` so that you don't need to arbitrarily choose one of them.